File: Day059_2025/Q059_GFG.py
class Solution:
    def evaluate(self, arr):
        st = []

        for token in arr:
            if token not in {"+", "-", "*", "/"}:
                st.append(int(token))
            else:
                x = st.pop()
                y = st.pop()

                if token == "+":
                    st.append(y + x)
                elif token == "-":
                    st.append(y - x)
                elif token == "*":
                    st.append(y * x)
                else:
                    st.append(int(y / x))  # Ensures integer division as in C++

        return st.pop()

File: Day059_2025/Q059_GFG.java
import java.util.Stack;

class Q059_GFG{
    public int evaluate(String[] arr) {
        Stack<Integer> st = new Stack<>();

        for (String token : arr) {
            if (!token.equals("+") && !token.equals("-") && !token.equals("*") && !token.equals("/")) {
                st.push(Integer.parseInt(token));
            } else {
                int x = st.pop();
                int y = st.pop();

                if (token.equals("+")) {
                    st.push(y + x);
                } else if (token.equals("-")) {
                    st.push(y - x);
                } else if (token.equals("*")) {
                    st.push(y * x);
                } else {
                    st.push(y / x);
                }
            }
        }

        return st.pop();
    }
}
